apps/hpc-ftsadmin/src/app/components/flow-form.tsx

```diff
@@ -0,0 +1,4094 @@
+          query: 'earmarkingType',
+        });
+      const category = fetchedCategories.filter(
+        (item) => item.name === 'Earmarked'
+      );
+      setFieldValue('earmarkingType', {
+        value: category[0],
+        displayLabel: category[0].name,
+      });
+      setObjectsWithArray(
+        publishedVersion,
+        [
+          'destinationPlans',
+          'destinationLocations',
+          'destinationGoverningEntities',
+          'destinationGlobalClusters',
+          'destinationOrganizations',
+        ],
+        [
+          'plans',
+          'locations',
+          'governingEntities',
+          'globalClusters',
+          'organizations',
+        ],
+        setFieldValue,
+        values
+      );
+    } else {
+      setObjectsWithArray(
+        publishedVersion,
+        [
+          'sourcePlans',
+          'sourceLocations',
+          'sourceGoverningEntities',
+          'sourceGlobalClusters',
+          'sourceOrganizations',
+        ],
+        [
+          'plans',
+          'locations',
+          'governingEntities',
+          'globalClusters',
+          'organizations',
+        ],
+        setFieldValue,
+        values
+      );
+    }
+  };
+  const fetchOrganizationDetails = async (
+    objectType: string,
+    organization: any,
+    setFieldValue: any,
+    values?: any
+  ) => {
+    const fetchedOrg =
+      await environment.model.organizations.getOrganizationsById(
+        organization[0].value
+      );
+    const isGovernment = (fetchedOrg[0].categories ?? []).some(
+      function (category) {
+        return (
+          category.group === 'organizationType' &&
+          [114, 123].includes(category.id)
+        );
+      }
+    );
+    if (isGovernment && objectType === 'sourceOrganizations') {
+      objects.sourceLocations = checkIfExistingAndCopy(
+        objects.location,
+        fetchedOrg[0],
+        'locations'
+      );
+      setObjectsWithArray(
+        objects,
+        ['sourceLocations'],
+        ['locations'],
+        setFieldValue,
+        values
+      );
+    }
+  };
+  const fetchAssociatedGoverningEntity = async (
+    objectType: string,
+    globalCluster: any,
+    setFieldValue: any,
+    values?: any
+  ) => {
+    if (
+      (values.sourcePlans.length === 0 &&
+        objectType === 'sourceGlobalClusters') ||
+      (values.destinationPlans.length === 0 &&
+        objectType === 'destinationGlobalClusters')
+    ) {
+      return;
+    }
+    let plan = null;
+    let targetGoverningEntities: any = null;
+    if (objectType === 'sourceGlobalClusters') {
+      plan = values.sourcePlans[0];
+      targetGoverningEntities = values.sourceGoverningEntities;
+    } else {
+      plan = values.destinationPlans[0];
+      targetGoverningEntities = values.destinationGoverningEntities;
+    }
+    const fetchedGoverningEntities =
+      await environment.model.governingEntities.getAllPlanGoverningEntities(
+        plan.value
+      );
```

Comment:
If you want to use let I would recommend you to type them, and there is no need for assigning an initial value
```suggestion
    let plan: Plan;
    let targetGoverningEntities: Array<GoverningEntity>;
    if (objectType === 'sourceGlobalClusters') {
      plan = values.sourcePlans[0];
      targetGoverningEntities = values.sourceGoverningEntities;
    } else {
      plan = values.destinationPlans[0];
      targetGoverningEntities = values.destinationGoverningEntities;
    }
    const fetchedGoverningEntities =
      await environment.model.governingEntities.getAllPlanGoverningEntities(
        plan.value
      );
```
I didn't use the proper types, they are just there for the example.